src/pools/pool.decorator.ts

```diff
@@ -0,0 +1,34 @@
+import { Pool, Token } from '../types';
+import { StaticPoolRepository, StaticTokenPriceProvider, Pool as SDKPool } from '@balancer-labs/sdk';
+import { tokensToTokenPrices } from '../tokens';
+import { PoolService } from './pool.service';
+
+
+
+export class PoolDecorator {
+  constructor(
+    public pools: Pool[]
+  ) {}
+
+  public async decorate(tokens: Token[]): Promise<Pool[]> {
+    console.log("------- START Decorating pools --------")
+    
+    const tokenPrices = tokensToTokenPrices(tokens);
+  
+    const poolProvider = new StaticPoolRepository(this.pools as SDKPool[]);
+    const tokenPriceProvider = new StaticTokenPriceProvider(tokenPrices);
+
+    const promises = this.pools.map(async pool => {
+      const poolService = new PoolService(pool, poolProvider, tokenPriceProvider);
+
+      await poolService.setTotalLiquidity();
+
+      return poolService.pool;
```

Comment:
Yea this is basically a thin wrapper around the SDK that adds logs and error handling. Once Jarek's APR PR is merged I'd like to use the new Pools model to add the data but this will look pretty similar. I just based this off what the frontend does for now.